Author: Konstantin Demin <rockdrilla@gmail.com>
Description: adjust build flags
Forwarded: no

debhelper and dpkg are fine enough in Debian (and derivatives) to setup flags.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -197,15 +197,6 @@ elseif(KEEPASSXC_DIST_TYPE STREQUAL "Other")
     unset(KEEPASSXC_DIST)
 endif()
 
-if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.14.0")
-    cmake_policy(SET CMP0083 NEW)
-    include(CheckPIESupported)
-    check_pie_supported()
-endif()
-
-# Create position independent code for shared libraries and executables
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-
 if("${CMAKE_SIZEOF_VOID_P}" EQUAL "4")
     set(IS_32BIT TRUE)
 endif()
@@ -300,14 +291,6 @@ if(CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
     add_gcc_compiler_flags("-Werror")
 endif()
 
-if (NOT HAIKU)
-    if((CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.8.999) OR CMAKE_COMPILER_IS_CLANGXX)
-        add_gcc_compiler_flags("-fstack-protector-strong")
-    else()
-        add_gcc_compiler_flags("-fstack-protector --param=ssp-buffer-size=4")
-    endif()
-endif()
-
 add_gcc_compiler_cxxflags("-Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual")
 add_gcc_compiler_cflags("-Wchar-subscripts -Wwrite-strings")
 
@@ -326,10 +309,6 @@ if(WITH_ASAN)
 
 endif()
 
-if(CMAKE_BUILD_TYPE_LOWER MATCHES "(release|relwithdebinfo|minsizerel)")
-    add_gcc_compiler_flags("-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2")
-endif()
-
 check_add_gcc_compiler_flag("-Werror=format-security")
 check_add_gcc_compiler_flag("-Werror=implicit-function-declaration" C)
 check_add_gcc_compiler_flag("-Wcast-align")
@@ -337,9 +316,7 @@ check_add_gcc_compiler_flag("-Wcast-align")
 if(UNIX AND NOT APPLE)
     check_add_gcc_compiler_flag("-Qunused-arguments")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--as-needed -Wl,--no-undefined")
-    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,relro,-z,now -pie")
     set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--as-needed")
-    set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,-z,relro,-z,now")
 endif()
 
 set(CMAKE_C_STANDARD 99)
